Fixed regression in application access dialing.

russjones opened the pull request:
**Description**

Pass the target address, in the case of application access `services.LocalNode`, in the Dial request to the reverse tunnel subsystem instead of filling it in within the reverse tunnel subsystem.

russjones committed:
Fixed regression in application access dialing.

Pass the target address, in the case of application access
services.LocalNode, in the Dial request to the reverse tunnel subsystem
instead of filling it in within the reverse tunnel subsystem.

awly commented on lib/web/app/transport.go:
```diff
@@ -158,11 +158,8 @@ func dialFunc(c *transportConfig) func(ctx context.Context, network string, addr
-			// debug within the reverse tunnel subsystem.
 			From:     &utils.NetAddr{AddrNetwork: "tcp", Addr: "@web-proxy"},
-			To:       &utils.NetAddr{AddrNetwork: "tcp", Addr: fmt.Sprintf("@app-%v", c.app.Name)},
+			To:       &utils.NetAddr{AddrNetwork: "tcp", Addr: reversetunnel.LocalNode},
```
Should there be a `reversetunnel.LocalApp` instead?
`LocalNode` means "local SSH server"

russjones replied:
That's what I originally had, but it was identical to `LocalNode`.

awly replied:
Maybe leave a TODO and add `LocalApp` to all the relevant switches in `lib/reversetunnel`.
Even if it's the same code, it's confusing to read here.

awly reviewed (commented):
Can you add a minimal regression test somewhere?
Seems like subtle behavior that can break again.

russjones commented:
@awly Planning on adding integration test coverage for this, but want to cut beta.11.

russjones committed:
Fixed migration version check logic.

Fixed migration version check logic to allow Application Access to work
on development builds.
